templates/question_with_approaches.jinja2

```diff
@@ -0,0 +1,197 @@
+{# templates/question_with_approaches #}
+---
+{% if question.all_tags %}
+tags:
+{% for tag in question.all_tags|sort %}
+  - {{ tag }}
+{% endfor %}
+{% endif %}
+
+hide:
+  - navigation
+  - tags
+---
+
+# {{ question.name }}
+
+**ID**: {{ question.id }}
+
+{% if question.description %}
+**Description**: {{ question.description }}
+{% endif %}
+
+**Approaches to Answer**
+
+{% for approach in question.approaches %}
+{% if components[approach].is_internal == False or allow_internal == True %}
+-  [{{ components[approach].name }} [{{ approach }}]](#{{ components[approach].name.replace(' ','-').lower() }})
+    - {{ components[approach]['description']['summary'] }}
+    - Tags: <span class="dfiqTag">{{ components[approach]['tags']|join('</span> <span class="dfiqTag">') }}</span>
+{%   endif  %}
+{% endfor %}
+
+# Approaches
+
+!!! Note
+    There are often multiple ways to approach answering a question.
+    This section explains the approaches considered, how they work, and any
+    benefits or drawbacks to each. All these approaches have different pros
+    and cons. They can be used individually or in conjunction.
+
+{% for approach in question.approaches %}
+{%   if components[approach].is_internal == False or allow_internal == True %}
+{% set a =  components[approach] %}
```

Comment:
Personally I prefer more descriptive naming, e.g:

```suggestion
{% for _approach in question.approaches %}
{%   if components[_approach].is_internal == False or allow_internal == True %}
{% set approach =  components[_approach] %}
```
Fine to leave as-is, but maybe something to consider.